pennylane_lightning_gpu/src/algorithms/ObservablesGPUMPI.hpp

```diff
@@ -404,4 +406,294 @@ class HamiltonianGPUMPI final : public ObservableGPUMPI<T> {
     }
 };
 
+/**
+ * @brief Sparse representation of HamiltonianGPUMPI<T>
+ *
+ * @tparam T Floating-point precision.
+ */
+template <typename T>
+class SparseHamiltonianGPUMPI final : public ObservableGPUMPI<T> {
+  public:
+    using PrecisionT = T;
+    // cuSparse required index type
+    using IdxT = typename std::conditional<std::is_same<T, float>::value,
+                                           int32_t, int64_t>::type;
+
+  private:
+    std::vector<std::complex<T>> data_;
+    std::vector<IdxT> indices_;
+    std::vector<IdxT> offsets_;
+    std::vector<std::size_t> wires_;
+
+    [[nodiscard]] bool
+    isEqual(const ObservableGPUMPI<T> &other) const override {
+        const auto &other_cast =
+            static_cast<const SparseHamiltonianGPUMPI<T> &>(other);
+
+        if (data_ != other_cast.data_ || indices_ != other_cast.indices_ ||
+            offsets_ != other_cast.offsets_) {
+            return false;
+        }
+
+        return true;
+    }
+
+  public:
+    /**
+     * @brief Create a SparseHamiltonianMPI from data, indices and offsets in
+     * CSR format.
+     *
+     * @param arg1 Arguments to construct data
+     * @param arg2 Arguments to construct indices
+     * @param arg3 Arguments to construct offsets
+     * @param arg4 Arguments to construct wires
+     */
+    template <typename T1, typename T2, typename T3 = T2,
```

Comment:
Should we not force `T2 == T3`? `wires` is also usually assumed to be of type `std::vector<size_t>`. In Lightning-Unification, sparse routine are usually templated as follows
```
template <class ComplexT, class index_type>
```
I'm realizing `index_type` in not upper camel case (maybe should be `IndexType` :D) Maybe we don't need  to template all argument types.

Reply:
Agree. We can probably work on it later when integrating L-GPU into the L-Unification. I'd prefer to keep the current implementation aligned with the non-MPI backend one. How do you think?

Reply:
Better keep it aligned intra-repo ;)